### PR TITLE
[INFRA] Add ESLint & Prettier 

### LINKED ISSUE #21: [FEAT] Add EsLint
See how it does in https://github.com/process-analytics/bpmn-visualization-js

### PR DESCRIPTION
**Pull request by csouchet**
Closes #21

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +48,18 @@
...
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-prettier": "^3.4.0",
+    "lint-staged": "^11.0.0",
+    "prettier": "^2.3.1",
     "typescript": "^4.3.4"
+  },
+  "lint-staged": {
+    "*.{js,ts,tsx}": [
+      "eslint --fix"
+    ]
```
Comment (tbouffard) [inline package.json:63]: ℹ️ probably not used

File: .github/workflows/build-pr.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Build PR
```
Comment (tbouffard) [inline .github/workflows/build-pr.yml:1]: 👍🏿 for splitting check from the site preview deploy

**Review by tbouffard — APPROVED** [review]
LGTM, some minor changes could be added

File: package.json (hunk 1 above)
Comment (csouchet) [reply package.json:63]: It's the same configuration that BPMN Visualization. 
It's used on commit, if you don't run the command to format the code.

File: .github/workflows/build-pr.yml (hunk 2 above)
Comment (csouchet) [reply .github/workflows/build-pr.yml:1]: I's almost the same as the job to deploy the site. Maybe we should merge the common steps in another workflow later.

File: package.json (hunk 1 above)
Comment (tbouffard) [reply package.json:64]: I haven't seen the husky configuration, so truly legit. :+1

**Pull request merged**